python/lsst/jointcal/jointcal.py

```diff
@@ -596,14 +597,17 @@ def _fit_photometry(self, associations, dataName=None):
         self.log.info("=== Starting photometric fitting...")
 
         # TODO: should use pex.config.RegistryField here (see DM-9195)
-        if self.config.photometryModel == "constrained":
+        if self.config.photometryModel == "constrainedFlux":
             model = lsst.jointcal.ConstrainedPhotometryModel(associations.getCcdImageList(),
                                                              self.focalPlaneBBox,
                                                              visitOrder=self.config.photometryVisitOrder)
             # potentially nonlinear problem, so we may need a line search to converge.
             doLineSearch = self.config.allowLineSearch
-        elif self.config.photometryModel == "simple":
-            model = lsst.jointcal.SimplePhotometryModel(associations.getCcdImageList())
+        elif self.config.photometryModel == "simpleFlux":
+            model = lsst.jointcal.SimpleFluxModel(associations.getCcdImageList())
+            doLineSearch = False  # purely linear problem, so no line search needed
```

Comment:
I don't think the simple flux model is linear - you have FittedStar parameters and observational parameters that get multiplied together, so it's linear in any *single* parameter, but for a problem to be linear it needs to be linear in the parameter *vector*.

Reply:
Good point. I'll clarify it to say that the model parameters are purely linear, which I'm pretty sure is the cause of our trouble and need for the lineSearch.